### PR TITLE
raidboss: Add P3N timeline

### PR DESCRIPTION
**Pull request by JLGarber**

### CODE REVIEW

File: ui/raidboss/data/06-ew/raid/p3n.txt [hunk 1]
```diff
@@ -0,0 +1,110 @@
+## Pandaemonium: Asphedolos, the Third Circle
+
+hideall "--Reset--"
+hideall "--sync--"
+
+0.0 "--Reset--" sync / 21:........:40000010:/ window 100000 jump 0
+
+# -ii 669F 66A7 66A8 66A0 66A1 66AF 66B0 66B3 6D0C 6DDC 6EDD
```
Comment (Legends0) [inline ui/raidboss/data/06-ew/raid/p3n.txt:8]: 66A5 (Joint Pyre) is missed?

File: ui/raidboss/data/06-ew/raid/p3n.txt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+86.1 "Brightened Fire #1" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
+87.2 "Brightened Fire #2" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
+88.4 "Brightened Fire #3" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
+89.5 "Brightened Fire #4" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
+105.6 "Left Cinderwing/Right Cinderwing" sync / 1[56]:[^:]*:Phoinix:(66B4|66B5):/
+
+# There might be an enrage here? More data will be needed.
+110.7 "--untargetable--"
+118.0 "Trail of Condemnation" sync / 1[56]:[^:]*:Phoinix:66AD:/window 118.0,10
+127.0 "Blazing Rain" sync / 1[56]:[^:]*:Phoinix:66A2:/
```
Comment (Legends0) [inline ui/raidboss/data/06-ew/raid/p3n.txt:32]: Adds are also targetable at this time point. May we add "--adds-targetable--" before?

File: ui/raidboss/data/06-ew/raid/p3n.txt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+hideall "--sync--"
+
+0.0 "--Reset--" sync / 21:........:40000010:/ window 100000 jump 0
+
+# -ii 669F 66A7 66A8 66A0 66A1 66AF 66B0 66B3 6D0C 6DDC 6EDD
+# -ic "Liturgic Bell"
+# -it Phoinix
+# -p 66AD:118 66A3: 200
+
+0.0 "--sync--" sync /Engage!/ window 0,1
```
Comment (Legends0) [inline ui/raidboss/data/06-ew/raid/p3n.txt:13]: We can sync on the first hit:
`3.8 "--sync--" sync / 1[56]:[^:]*:Phoinix:368:/ window 4.3,0`

**Comment by Legends0** [comment]
Is it worth breaking apart the fireplume in/out?

Example:
```
13.8 "Experimental Fireplume Out/Middle?" sync / 1[56]:[^:]*:Phoinix:6696:/ window 14.4,5 jump 213.8
13.9 "--sync--" sync / 1[56]:[^:]*:Phoinix:6698:/ window 14.4,5 jump 113.9
19.1 "Fireplume 1?" #sync / 1[56]:[^:]*:Phoinix:6699:/
20.0 "Fireplume 2?" #sync / 1[56]:[^:]*:Phoinix:6699:/
21.0 "Fireplume 3?" #sync / 1[56]:[^:]*:Phoinix:6699:/
22.0 "Fireplume 4?" #sync / 1[56]:[^:]*:Phoinix:6699:/
23.0 "Fireplume Middle?" #sync / 1[56]:[^:]*:Phoinix:6699:/
31.0 "Experimental Fireplume Middle/Out?" #sync / 1[56]:[^:]*:Phoinix:6696:/
37.1 "Fireplume Middle?" #sync / 1[56]:[^:]*:Phoinix:6697:/
49.2 "Scorched Exaltation" #sync / 1[56]:[^:]*:Phoinix:66B8:/ window 150,10
61.4 "--sync--" #sync / 1[56]:[^:]*:Phoinix:66B2:/
62.4 "Heat of Condemnation" #sync / 1[56]:[^:]*:Phoinix:66B3:/

# Fireplume Out First
113.9 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:6698:/
119.1 "Fireplume 1" sync / 1[56]:[^:]*:Phoinix:6699:/
120.0 "Fireplume 2" sync / 1[56]:[^:]*:Phoinix:6699:/
121.0 "Fireplume 3" sync / 1[56]:[^:]*:Phoinix:6699:/
122.0 "Fireplume 4" sync / 1[56]:[^:]*:Phoinix:6699:/
123.0 "Fireplume Middle" sync / 1[56]:[^:]*:Phoinix:6699:/
131.0 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:6696:/
137.1 "Fireplume Middle" sync / 1[56]:[^:]*:Phoinix:6697:/
149.2 "Scorched Exaltation" #sync / 1[56]:[^:]*:Phoinix:66B8:/ window 150,10
161.4 "--sync--" #sync / 1[56]:[^:]*:Phoinix:66B2:/
162.4 "Heat of Condemnation" #sync / 1[56]:[^:]*:Phoinix:66B3:/
175.6 "Darkened Fire" #sync / 1[56]:[^:]*:Phoinix:6EDA:/
185.7 "--sync--" #sync / 1[56]:[^:]*:Phoinix:6EDC:/
186.1 "Brightened Fire 1" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
187.2 "Brightened Fire 2" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
188.4 "Brightened Fire 3" #sync / 1[56]:[^:]*:Phoinix:6EDD:/
189.6 "Brightened Fire 4" #sync / 1[56]:[^:]*:Phoinix:6EDD:/

# Fireplume Middle First
213.8 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:6696:/
219.9 "Fireplume Middle" sync / 1[56]:[^:]*:Phoinix:6697:/
231.0 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:6698:/
236.2 "Fireplume 1" sync / 1[56]:[^:]*:Phoinix:6699:/
237.1 "Fireplume 2" sync / 1[56]:[^:]*:Phoinix:6699:/
238.1 "Fireplume 3" sync / 1[56]:[^:]*:Phoinix:6699:/
239.1 "Fireplume 4" sync / 1[56]:[^:]*:Phoinix:6699:/
240.1 "Fireplume Middle" sync / 1[56]:[^:]*:Phoinix:6699:/
```

File: ui/raidboss/data/06-ew/raid/p3n.txt (hunk 3 above)
Comment (quisquous) [reply ui/raidboss/data/06-ew/raid/p3n.txt:13]: I find initial auto syncs to often be very inconsistent.  I think better to just have a starts casting on the first ability, for consistency.

File: ui/raidboss/data/06-ew/raid/p3n.txt (hunk 1 above)
Comment (JLGarber) [reply ui/raidboss/data/06-ew/raid/p3n.txt:8]: I was working off a timeline  where I had initially ignored all Sparkfledged adds, so I missed putting it here. Re-added

**Comment by JLGarber** [comment]
I'm not sure whether it is. The timing on all other mechanics seems unchanged regardless of which pattern we get, and I feel kind of weird dedicating two whole blocks to a single mechanic that doesn't change any timings.

**Comment by quisquous** [comment]
> I'm not sure whether it is. The timing on all other mechanics seems unchanged regardless of which pattern we get, and I feel kind of weird dedicating two whole blocks to a single mechanic that doesn't change any timings.

I think it makes sense not to add a block here.  However, with my melee hat on, I think it'd be useful to know when the "middle" hit? Is that the timing of the fireplume that's listed? Or, would it be confusing to say `--middle?--` on both branches?

**Comment by JLGarber** [comment]
That's fair. Here's what I have:

```
0.0 "--sync--" sync /Engage!/ window 0,1
13.9 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:6696:/
20.0 "Fireplume" sync / 1[56]:[^:]*:Phoinix:6697:/
```
```
0.0 "--sync--" sync /Engage!/ window 0,1
14.0 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:6698:/
19.2 "Fireplume" sync / 1[56]:[^:]*:Phoinix:6699:/
20.1 "Fireplume" sync / 1[56]:[^:]*:Phoinix:6699:/
21.1 "Fireplume" sync / 1[56]:[^:]*:Phoinix:6699:/
22.1 "Fireplume" sync / 1[56]:[^:]*:Phoinix:6699:/
23.1 "Fireplume" sync / 1[56]:[^:]*:Phoinix:6699:/
```

Re: labeling the center hit, I'm not quite sure how we would. Maybe we could do a monstrosity like this?
```
0.0 "--sync--" sync /Engage!/ window 0,1
14.0 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:(6696|6698):/
19.2 "Optical Fireplume?" #sync / 1[56]:[^:]*:Phoinix:6699:/
20.1 "Chariot Fireplume?" sync / 1[56]:[^:]*:Phoinix:6697:/
20.1 "Optical Fireplume?" #sync / 1[56]:[^:]*:Phoinix:6699:/
21.1 "Optical Fireplume?" #sync / 1[56]:[^:]*:Phoinix:6699:/
22.1 "Optical Fireplume?" #sync / 1[56]:[^:]*:Phoinix:6699:/
23.1 "Optical Fireplume (center)?" #sync / 1[56]:[^:]*:Phoinix:6699:/
```

**Comment by quisquous** [comment]
Subjectively, I think we could drop the Optical Fireplume, and just leave "Chariot Fireplume?".  If you think it's too confusing, I'm fine with not having either.

File: ui/raidboss/data/06-ew/raid/p3n.txt [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+253.8 "Ashen Eye #1" #sync / 1[56]:[^:]*:Sparkfledged:66AB:/
+255.7 "Ashen Eye #2" #sync / 1[56]:[^:]*:Sparkfledged:66AB:/
+257.7 "Experimental Charplume" sync / 1[56]:[^:]*:Phoinix:669C:/
+257.7 "Ashen Eye #3" #sync / 1[56]:[^:]*:Sparkfledged:66AB:/
+259.7 "Ashen Eye #4" #sync / 1[56]:[^:]*:Sparkfledged:66AB:/
+262.8 "Charplume" sync / 1[56]:[^:]*:Phoinix:669D:/
+262.8 "--sync--" sync / 1[56]:[^:]*:Phoinix:66AC:/
+277.9 "Heat of Condemnation" sync / 1[56]:[^:]*:Phoinix:66B2:/
+289.2 "--sync--" sync / 1[56]:[^:]*:Phoinix:669E:/
+291.3 "Devouring Brand" sync / 1[56]:[^:]*:Phoinix:66A0:/ duration 13
```
Comment (Legends0) [inline ui/raidboss/data/06-ew/raid/p3n.txt:62]: Can we add the time for a "Devouring Brand's Expands"? I think it's the 6D82 id.

**Comment by JLGarber** [comment]
Maybe something like this for each Fireplume sequence?

```
0.0 "--sync--" sync /Engage!/ window 0,1
14.0 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:(6696|6698):/
20.1 "--center fireplume?--" sync / 1[56]:[^:]*:Phoinix:6697:/
23.1 "--center fireplume?--" #sync / 1[56]:[^:]*:Phoinix:6699:/
```

**Comment by quisquous** [comment]
> Maybe something like this for each Fireplume sequence?
> 
> ```
> 0.0 "--sync--" sync /Engage!/ window 0,1
> 14.0 "Experimental Fireplume" sync / 1[56]:[^:]*:Phoinix:(6696|6698):/
> 20.1 "--center fireplume?--" sync / 1[56]:[^:]*:Phoinix:6697:/
> 23.1 "--center fireplume?--" #sync / 1[56]:[^:]*:Phoinix:6699:/
> ```

Hmm.  So, what I really wanted was "what is the timing of the quick telegraphed middle section I need to get out for".  Having two middles makes it unclear which one to get out for.  At this point, I guess I'd prefer that we just have one "center fireplume?" for the "get out" version or no fireplumes at all.

**Comment by JLGarber** [comment]
I guess it comes down to "which is more valuable, knowing when the giant fireplume is, or avoiding confusion between Chariot and Optical Sight?" I feel like `Giant Fireball?` is probably reasonably non-confusing, since people will already have a trigger to warn them if it's Optical Sight, and that doesn't require precise snapshot timing.

**Comment by quisquous** [comment]
> I guess it comes down to "which is more valuable, knowing when the giant fireplume is, or avoiding confusion between Chariot and Optical Sight?" I feel like `Giant Fireball?` is probably reasonably non-confusing, since people will already have a trigger to warn them if it's Optical Sight, and that doesn't require precise snapshot timing.

Yeah, that's my reasoning as well.

File: ui/raidboss/data/06-ew/raid/p3n.txt (hunk 4 above)
Comment (JLGarber) [reply ui/raidboss/data/06-ew/raid/p3n.txt:62]: Good catch, thanks for the suggestion.

**Pull request merged**